terminus-ui/expansion-panel/src/trigger/expansion-panel-trigger.component.ts

```diff
@@ -130,18 +130,20 @@ export class TsExpansionPanelTriggerComponent implements OnDestroy, FocusableOpt
     @Inject(TS_EXPANSION_PANEL_DEFAULT_OPTIONS) @Optional() defaultOptions?: TsExpansionPanelDefaultOptions,
   ) {
     const accordionHideToggleChange =
-      panel.accordion ?
-      panel.accordion._stateChanges.pipe(filter((changes) => !!changes['hideToggle'])) :
-      EMPTY;
+      panel.accordion
+        // NOTE: Underscore naming controlled by Material
+        // eslint-disable-next-line no-underscore-dangle
+        ? panel.accordion._stateChanges.pipe(filter(changes => !!changes.hideToggle))
+        : EMPTY;
 
     // Since the toggle state depends on an @Input on the panel, we need to subscribe and trigger change detection manually.
     merge(
       panel.opened, panel.closed, accordionHideToggleChange,
-      panel.inputChanges.pipe(filter((changes) => !!(changes['hideToggle'] || changes['disabled']))),
+      panel.inputChanges.pipe(filter(changes => !!(changes.hideToggle || changes.disabled))),
```

Comment:
Yep, there is a rule called `dot-notation` that enforces this.

The dot notation is often preferred because it is easier to read, less verbose, and works better with aggressive JavaScript minimizers. Then on top of that, for TypeScript if `--noImplicitAny` is turned off, property access via a string literal will be ‘any’ if the property does not exist.

Reply:
This rule is disabled for spec files since we often need to use the bracket approach to access private items.